src/rgw/rgw_cr_rest.h

```diff
@@ -35,6 +39,7 @@ class RGWReadRESTResourceCR : public RGWSimpleCoroutine {
     if (ret < 0) {
       log_error() << "failed to send http operation: " << op->to_str()
           << " ret=" << ret << std::endl;
+      op->put();
```

Comment:
or, if we always use intrusive_ptr, change the object to init the ref 
count at 0.  (That's what we usually do.)

On Wed, 4 May 2016, Casey Bodley wrote:

> 
> In src/rgw/rgw_cr_rest.h:
> 
> > @@ -35,6 +39,7 @@ class RGWReadRESTResourceCR : public RGWSimpleCoroutine {
> >      if (ret < 0) {
> >        log_error() << "failed to send http operation: " << op->to_str()
> >            << " ret=" << ret << std::endl;
> > +      op->put();
> 
> oh, gross! intrusive_ptr()'s ctor takes a second bool add_ref = true argument, let's pass false to keep it at 1 then
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub[AAabh-0U8_ruIeyr6B3q2Dmc9aMYlNM0ks5p-K8NgaJpZM4IWpG4.gif]
> 
> 
>